### PR TITLE
feat(SDK): add HyperealVR SDK support

### PR DESCRIPTION
**Pull request by crmxm**
The pull request implements the HyperealVR SDK for VRTK.

The setup is pretty strightforward:

1. Import Hypereal VR Unity Plugin (https://www.hypereal.com/#/developer/resource/download)
2. Drag HyperealVRMain prefab into the scene.
3. Add VRTK_SDKSetup script onto the HyperealVRMain and choose Hypereal SDK in SDK selection.
4. Add the setup into SDKManager.

Notice:
1. For some versions of plugin, HyPointerLaser script is attached on the controllers. Remove them before use.

2. Since we only have one button on each controller, we could not map all VRTK buttons. The menu button on the right controller is mapped to VRTK button-menu while the menu button on the  left is mapped to button-two of VRTK. VRTK button-one has not been mapped to any button of the controller.

3. HyperealVR setting menu will pop out when double clicking the menu button on the right. Currently, there's issue with double clicking. Our engineer is working on that and the fixed SDK is releasing next Tuesday (July 18th).

### CODE REVIEW

**Comment by bddckr** [comment]
Hey there! I suggest also editing the [GETTING_STARTED](https://github.com/thestonefox/VRTK/blob/master/GETTING_STARTED.md) document to include the above description. Look at the already included descriptions for the other SDKs to see the wording used. Make sure to link the Unity plugin there, too. 😄

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 4643f3b0b40a9b1c9a844a4915d2196ec289f7e6
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/thestonefox/SteamVR_Unity_Toolkit

---
This message was auto-generated by https://gitcop.com

**Comment by bddckr** [comment]
Just squash the commits and make sure the resulting single commit's commit message is the one you used for the first commit, then force push the branch. That will update the PR here 😄

**Comment by crmxm** [comment]
Oh I forget to close the details tag. I will update it now.

**Comment by bddckr** [comment]
Ohh! Interestingly it just showed up fine when asking GitHub to render the markdown document. 🤷‍♂️

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRDefines.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    /// <summary>
+    /// Handles all the scripting define symbols for the Hypereal SDK.
+    /// </summary>
+    public static class SDK_HyperealVRDefines
+    {
+        /// <summary>
+        /// The scripting define symbol for the Hypereal SDK.
+        /// </summary>
+        public const string ScriptDefineSymbol = SDK_ScriptingDefineSymbolPredicateAttribute.RemovableSymbolPrefix +
```
Comment (bddckr) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRDefines.cs]: Small thing: We name this `ScriptingDefineSymbol` in all the other files. 😅

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRDefines.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            "SDK_HYPEREALVR";
+
+        private const string BuildTargetGroupName = "Standalone";
+
+        [SDK_ScriptingDefineSymbolPredicate(ScriptDefineSymbol, BuildTargetGroupName)]
+        [SDK_ScriptingDefineSymbolPredicate(SDK_ScriptingDefineSymbolPredicateAttribute.RemovableSymbolPrefix +
```
Comment (bddckr) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRDefines.cs]: No need for this attribute when you don't have multiple versions to add scripting define symbols for _in addition to the general SDK symbol_. So I suggest to remove this line. (Also see the next comment.)

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRDefines.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            "HYPEREALVR_PLUGIN_1_0_OR_NEWER", BuildTargetGroupName)]
+        private static bool IsPluginVersion10OrNewer()
```
Comment (bddckr) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRDefines.cs]: No need for versioned symbol, so rename to `IsHyperealAvailable` instead. [See the Ximmerse defines file for an example.](https://github.com/thestonefox/VRTK/blob/master/Assets/VRTK/SDK/Ximmerse/SDK_XimmerseDefines.cs#L16)

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRDefines.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            return true;
```
Comment (bddckr) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRDefines.cs]: This actually needs to check for existance of any of the types from the Hypereal SDK. Use `VRTK_SharedMethods.GetTypeUnknownAssembly` and make sure to use a namespace-prefixed type name for the argument. [See the Ximmerse defines file for an example.](https://github.com/thestonefox/VRTK/blob/master/Assets/VRTK/SDK/Ximmerse/SDK_XimmerseDefines.cs#L18)

**Comment by crmxm** [comment]
`Transform GetHeadset()`, `Transform GetHeadsetCamera()`, `Transform GetPlayArea()` function changed.
Previous implementation may cause error when component not found.

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRBoundaries.cs [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        /// The GetPlayArea method returns the Transform of the object that is used to represent the play area in the scene.
+        /// </summary>
+        /// <returns>A transform of the object representing the play area in the scene.</returns>
+        public override Transform GetPlayArea()
+        {
+            cachedPlayArea = GetSDKManagerPlayArea();
+            if (cachedPlayArea == null)
+            {
+                GameObject myHeadGO = VRTK_SharedMethods.FindEvenInactiveGameObject<HyHead>();
+                cachedPlayArea = myHeadGO != null ? myHeadGO.transform.parent : null;
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRBoundaries.cs]: wrap this in brackets to make it consistent with other uses of this pattern.

`cachedPlayArea = (myHeadGO != null ? myHeadGO.transform.parent : null);`

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 6]
```diff
@@ -0,0 +1,553 @@
...
+        }
+
+        /// <summary>
+        /// The GetControllerDefaultColliderPath returns the path to the prefab that contains the collider objects for the default controller of this SDK.
+        /// </summary>
+        /// <param name="hand">The controller hand to check for</param>
+        /// <returns>A path to the resource that contains the collider GameObject.</returns>
+        public override string GetControllerDefaultColliderPath(ControllerHand hand)
+        {
+            return "ControllerColliders/Fallback";
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: Isn't there a Hyperreal controller? the fallback collider is just a sphere, is that good enough for the Hyperreal experience?

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 7]
```diff
@@ -0,0 +1,553 @@
...
+        /// <summary>
+        /// The GetControllerElementPath returns the path to the game object that the given controller element for the given hand resides in.
+        /// </summary>
+        /// <param name="element">The controller element to look up.</param>
+        /// <param name="hand">The controller hand to look up.</param>
+        /// <param name="fullPath">Whether to get the initial path or the full path to the element.</param>
+        /// <returns>A string containing the path to the game object that the controller element resides in.</returns>
+        public override string GetControllerElementPath(ControllerElements element, ControllerHand hand, bool fullPath = false)
+        {
+            return null;
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: Again, if there is a controller this should be filled in so additional controller features are available, such as element highlighting and auto retrieval of the default attach point.

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 8]
```diff
@@ -0,0 +1,553 @@
...
+        /// The GetControllerLeftHand method returns the GameObject containing the representation of the left hand controller.
+        /// </summary>
+        /// <param name="actual">If true it will return the actual controller, if false it will return the script alias controller GameObject.</param>
+        /// <returns>The GameObject containing the left hand controller.</returns>
+        public override GameObject GetControllerLeftHand(bool actual = false)
+        {
+            if (actual)
+            {
+                HyTrackObjRig trackedObjRig = VRTK_SharedMethods.FindEvenInactiveComponent<HyTrackObjRig>();
+                if (trackedObjRig)
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: Need to wrap this in brackets:

```
if (trackedObjRig)
{
  return trackedObjRig.leftController;
}
```

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 9]
```diff
@@ -0,0 +1,553 @@
...
+        /// <summary>
+        /// The GetControllerByIndex method returns the GameObject of a controller with a specific index.
+        /// </summary>
+        /// <param name="index">The index of the controller to find.</param>
+        /// <param name="actual">If true it will return the actual controller, if false it will return the script alias controller GameObject.</param>
+        /// <returns>The GameObject of the controller</returns>
+        public override GameObject GetControllerByIndex(uint index, bool actual = false)
+        {
+            SetTrackedControllerCaches();
+            var sdkManager = VRTK_SDKManager.instance;
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: Don't use `var` use `VRTK_SDKManager sdkManager =` it just looks more consistent with other variable declarations

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 10]
```diff
@@ -0,0 +1,553 @@
...
+        {
+            if (actual)
+            {
+                HyTrackObjRig trackedObjRig = VRTK_SharedMethods.FindEvenInactiveComponent<HyTrackObjRig>();
+                if (trackedObjRig)
+                    return trackedObjRig.leftController;
+            }
+            else
+            {
+                var sdkManager = VRTK_SDKManager.instance;
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: use `VRTK_SDKManager sdkManager = `

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 11]
```diff
@@ -0,0 +1,553 @@
...
+        /// The GetControllerRightHand method returns the GameObject containing the representation of the right hand controller.
+        /// </summary>
+        /// <param name="actual">If true it will return the actual controller, if false it will return the script alias controller GameObject.</param>
+        /// <returns>The GameObject containing the right hand controller.</returns>
+        public override GameObject GetControllerRightHand(bool actual = false)
+        {
+            if (actual)
+            {
+                HyTrackObjRig trackedObjRig = VRTK_SharedMethods.FindEvenInactiveComponent<HyTrackObjRig>();
+                if (trackedObjRig)
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: brackets

```
if (trackedObjRig)
{
  return trackedObjRig.rightController;
}
```

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 12]
```diff
@@ -0,0 +1,553 @@
...
+                    return trackedObjRig.rightController;
+            }
+            else
+            {
+                var sdkManager = VRTK_SDKManager.instance;
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: `VRTK_SDKManager sdkManager = `

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 13]
```diff
@@ -0,0 +1,553 @@
...
+        }
+
+        /// <summary>
+        /// The GetControllerModel method returns the model alias for the given controller hand.
+        /// </summary>
+        /// <param name="hand">The hand enum of which controller model to retrieve.</param>
+        /// <returns>The GameObject that has the model alias within it.</returns>
+        public override GameObject GetControllerModel(ControllerHand hand)
+        {
+            var model = GetSDKManagerControllerModelForHand(hand);
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: Don't use `var` I'm guessing `model` is supposed to be of type GameObject?

if so, then use `GameObject model = `

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 14]
```diff
@@ -0,0 +1,553 @@
...
+                        controller = GetControllerLeftHand(true);
+                        break;
+                    case ControllerHand.Right:
+                        controller = GetControllerRightHand(true);
+                        break;
+                }
+
+                if (controller != null)
+                {
+                    model = controller.transform.Find("Model").gameObject;
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: Split this line into two as if `Model` is not found then you'll get null errors

```
Transform foundModel = controller.transform.Find("Model");
model = (foundModel != null ? foundModel.gameObject : null);
```

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 15]
```diff
@@ -0,0 +1,553 @@
...
+        /// The HapticPulse/2 method is used to initiate a simple haptic pulse on the tracked object of the given controller reference.
+        /// </summary>
+        /// <param name="controllerReference">The reference to the tracked object to initiate the haptic pulse on.</param>
+        /// <param name="strength">The intensity of the rumble of the controller motor. `0` to `1`.</param>
+        public override void HapticPulse(VRTK_ControllerReference controllerReference, float strength = 0.5f)
+        {
+            uint index = VRTK_ControllerReference.GetRealIndex(controllerReference);
+            HyDevice ctrlDevice = MappingIndex2HyDevice(index);
+
+            if (ctrlDevice == HyDevice.Device_Unknown)
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: Brackets

```
if (ctrlDevice == HyDevice.Device_Unknown)
{
  return;
}
```

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 16]
```diff
@@ -0,0 +1,553 @@
...
+        /// <summary>
+        /// The GetVelocity method is used to determine the current velocity of the tracked object on the given controller reference.
+        /// </summary>
+        /// <param name="controllerReference">The reference to the tracked object to check for.</param>
+        /// <returns>A Vector3 containing the current velocity of the tracked object.</returns>
+        public override Vector3 GetVelocity(VRTK_ControllerReference controllerReference)
+        {
+            uint index = VRTK_ControllerReference.GetRealIndex(controllerReference);
+            SetTrackedControllerCaches();
+            var sdkManager = VRTK_SDKManager.instance;
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: update to `VRTK_SDKManager sdkManager = `

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 17]
```diff
@@ -0,0 +1,553 @@
...
+        /// <summary>
+        /// The GetAngularVelocity method is used to determine the current angular velocity of the tracked object on the given controller reference.
+        /// </summary>
+        /// <param name="controllerReference">The reference to the tracked object to check for.</param>
+        /// <returns>A Vector3 containing the current angular velocity of the tracked object.</returns>
+        public override Vector3 GetAngularVelocity(VRTK_ControllerReference controllerReference)
+        {
+            uint index = VRTK_ControllerReference.GetRealIndex(controllerReference);
+            SetTrackedControllerCaches();
+            var sdkManager = VRTK_SDKManager.instance;
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: update to `VRTK_SDKManager sdkManager = `

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 18]
```diff
@@ -0,0 +1,553 @@
...
+        /// The GetButtonAxis method retrieves the current X/Y axis values for the given button type on the given controller reference.
+        /// </summary>
+        /// <param name="buttonType">The type of button to check for the axis on.</param>
+        /// <param name="controllerReference">The reference to the controller to check the button axis on.</param>
+        /// <returns>A Vector2 of the X/Y values of the button axis. If no axis values exist for the given button, then a Vector2.Zero is returned.</returns>
+        public override Vector2 GetButtonAxis(ButtonTypes buttonType, VRTK_ControllerReference controllerReference)
+        {
+            uint index = VRTK_ControllerReference.GetRealIndex(controllerReference);
+            HyDevice ctrlDevice = MappingIndex2HyDevice(index);
+            if (ctrlDevice == HyDevice.Device_Unknown)
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: brackets

```
if (ctrlDevice == HyDevice.Device_Unknown)
{
  return Vector2.zero;
}
```

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 19]
```diff
@@ -0,0 +1,553 @@
...
+        /// The GetButtonHairlineDelta method is used to get the difference between the current button press and the previous frame button press.
+        /// </summary>
+        /// <param name="buttonType">The type of button to get the hairline delta for.</param>
+        /// <param name="controllerReference">The reference to the controller to get the hairline delta for.</param>
+        /// <returns>The delta between the button presses.</returns>
+        public override float GetButtonHairlineDelta(ButtonTypes buttonType, VRTK_ControllerReference controllerReference)
+        {
+            uint index = VRTK_ControllerReference.GetRealIndex(controllerReference);
+            HyDevice ctrlDevice = MappingIndex2HyDevice(index);
+            if (ctrlDevice == HyDevice.Device_Unknown)
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: brackets

```
if (ctrlDevice == HyDevice.Device_Unknown)
{
  return 0f;
}
```

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 20]
```diff
@@ -0,0 +1,553 @@
...
+        /// </summary>
+        /// <param name="buttonType">The type of button to check for the state of.</param>
+        /// <param name="pressType">The button state to check for.</param>
+        /// <param name="controllerReference">The reference to the controller to check the button state on.</param>
+        /// <returns>Returns true if the given button is in the state of the given press type on the given controller reference.</returns>
+        public override bool GetControllerButtonState(ButtonTypes buttonType, ButtonPressTypes pressType, VRTK_ControllerReference controllerReference)
+        {
+            uint index = VRTK_ControllerReference.GetRealIndex(controllerReference);
+            HyDevice ctrlDevice = MappingIndex2HyDevice(index);
+            if (ctrlDevice == HyDevice.Device_Unknown)
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: brackets

```
if (ctrlDevice == HyDevice.Device_Unknown)
{
  return false;
}
```

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 21]
```diff
@@ -0,0 +1,553 @@
...
+                return false;
+
+            switch (buttonType)
+            {
+                case ButtonTypes.ButtonOne:
+                    return false;
+                case ButtonTypes.ButtonTwo:
+                    if (ctrlDevice == HyDevice.Device_Controller0)
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: brackets

```
if (ctrlDevice == HyDevice.Device_Controller0)
{
  return IsButtonPressed(ctrlDevice, pressType, HyInputKey.Menu);
}
else
{
  return false;
}
```

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 22]
```diff
@@ -0,0 +1,553 @@
...
+                        return IsButtonPressed(ctrlDevice, pressType, HyInputKey.Menu);
+                    else
+                        return false;
+                case ButtonTypes.Grip:
+                    return IsButtonPressed(ctrlDevice, pressType, HyInputKey.SideTrigger);
+                case ButtonTypes.GripHairline:
+                    return false;
+                case ButtonTypes.StartMenu:
+                    if (ctrlDevice == HyDevice.Device_Controller0)
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: brackets

```
if (ctrlDevice == HyDevice.Device_Controller0)
{
  return false;
}
else
{
  return IsButtonPressed(ctrlDevice, pressType, HyInputKey.Menu);
}
```

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 23]
```diff
@@ -0,0 +1,553 @@
...
+
+        private void SetTrackedControllerCaches(bool forceRefresh = false)
+        {
+            if (forceRefresh)
+            {
+                cachedLeftTrackedObject = null;
+                cachedRightTrackedObject = null;
+            }
+
+            var sdkManager = VRTK_SDKManager.instance;
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: update to `VRTK_SDKManager sdkManager = `

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 24]
```diff
@@ -0,0 +1,553 @@
...
+            if (sdkManager != null)
+            {
+                if (cachedLeftTrackedObject == null && sdkManager.loadedSetup.actualLeftController)
+                {
+                    cachedLeftController = sdkManager.loadedSetup.actualLeftController.GetComponent<VRTK_TrackedController>();
+                    if (cachedLeftController != null)
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: brackets

```
if (cachedLeftController != null)
{
  cachedLeftController.index = 0;
}
```

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 25]
```diff
@@ -0,0 +1,553 @@
...
+                        cachedLeftController.index = 0;
+                }
+                if (cachedRightTrackedObject == null && sdkManager.loadedSetup.actualRightController)
+                {
+                    cachedRightController = sdkManager.loadedSetup.actualRightController.GetComponent<VRTK_TrackedController>();
+                    if (cachedRightController != null)
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: brackets

```
if (cachedRightController != null)
{
  cachedRightController.index = 1;
}
```

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs [hunk 26]
```diff
@@ -0,0 +1,553 @@
...
+                default:
+                    break;
+            }
+            return HyDevice.Device_Unknown;
+        }
+
+        private bool IsButtonPressed(HyDevice index, ButtonPressTypes type, HyInputKey button)
+        {
+            //todo use index to input type
+            var device = HyInputManager.Instance.GetInputDevice(index);
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRController.cs]: replace `var` with the actual type (I'm not sure of what type that will return)

File: Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRHeadset.cs [hunk 27]
```diff
@@ -0,0 +1,122 @@
...
+            return false;
+        }
+
+        /// <summary>
+        /// The AddHeadsetFade method attempts to add the fade functionality to the game object with the camera on it.
+        /// </summary>
+        /// <param name="camera">The Transform to with the camera on to add the fade functionality to.</param>
+        public override void AddHeadsetFade(Transform camera)
+        {
+            if (camera && !camera.GetComponent<VRTK_ScreenFade>())
```
Comment (thestonefox) [inline Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRHeadset.cs]: make the `camera` null comparison explicit so:

`if (camera != null && !camera.GetComponent<VRTK_ScreenFade>())`

**Comment by thestonefox** [comment]
It's probably also worth updating the `README.md` document and adding HyperRealVR to the Supported SDK table

I'd also suggest marking it as experimental as no one has confirmed this SDK is working other than the author of the PR.

**Comment by crmxm** [comment]
I found that there's new a new method ` public abstract float GetButtonSenseAxis(ButtonTypes buttonType, VRTK_ControllerReference controllerReference);`. I will implement it and submit the changes later.

**Comment by crmxm** [comment]
Need I update the examples and `SDKSetups` prefab?

**Comment by bddckr** [comment]
Yeah we include all the supported SDKs in that prefab which is then used in the example scenes.

Previously there were some issues with that prefab, so one needed to only edit it with all the vendor SDKs imported... I think that has been fixed, though, because the SDK Setups auto-populate in their `OnEnable` these days. May be good to check with @thestonefox after you edited the prefab 😄

**Comment by crmxm** [comment]
I forget to update the readme.md.
Sorry about that.

**Comment by thestonefox** [comment]
@crmxm Double check your SDK still compiles with out errors as there have been more methods added to the SDK since this PR:

e.g. 
https://github.com/thestonefox/VRTK/blob/release/3.3.0-alpha/Assets/VRTK/SDK/Base/SDK_BaseController.cs#L264

**Comment by crmxm** [comment]
I updated the code.  It is up-to-date with the current release branch, and it works fine.

**Comment by thestonefox** [comment]
all looks good to me now. @bddckr do you want to have a final look over as well?

**Comment by bddckr** [comment]
Looks good to me, except I didn't check the files' first line. Dunno if the used doc generator number there is correct.

**Comment by thestonefox** [comment]
Ah well, if the docs dont auto gen properly, i'll fix that stuff in the auto gen doc commit ;)

**Pull request merged**